nRFMeshProvision/Classes/Mesh Messages/TimeMessage.swift

```diff
@@ -0,0 +1,129 @@
+/*
+* Copyright (c) 2022, Nordic Semiconductor
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without modification,
+* are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+*    list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright notice, this
+*    list of conditions and the following disclaimer in the documentation and/or
+*    other materials provided with the distribution.
+*
+* 3. Neither the name of the copyright holder nor the names of its contributors may
+*    be used to endorse or promote products derived from this software without
+*    specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+* NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+* PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+* WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import Foundation
+
+
+public protocol TimeMessage: StaticMeshMessage {
+    var time: TaiTime { get }
+}
+
+public struct TaiTime {
+    /// The current TAI time in seconds.
+    public let seconds: UInt64
+    /// The sub-second time in units of 1/256th second.
+    public let subSecond: UInt8
+    /// The estimated uncertainty in 10-millisecond steps.
+    public let uncertainty: UInt8
+    /// Whether this time is authorative (from a "known good" source, such as GPS or NTP).
+    public let authority: Bool
+    /// Current difference between TAI and UTC in seconds (range -255 to 32512).
+    public let taiDelta: Int16
+    /// The Local time zone offset.
```

Comment:
What is the unit here? Hours?

Reply:
Which property? taiDelta is seconds, like the comment says. Time offset was Double hours, but is changed to a TimeZone struct instead now.